src/services/sync/strategies/consumerActions/BaseConsumerAction.js

```diff
@@ -0,0 +1,80 @@
+const { NotImplemented, SyncError, BadRequest } = require('../../../../errors');
+const { batchFilterKeys } = require('../../utils');
+
+const securedKeys = ['type', 'getType', 'matchType', 'filterData', 'exec', 'action', 'getSecuredKeys'];
+
+class BaseConsumerAction {
+	constructor(type, options) {
+		if (!type) {
+			throw new NotImplemented('The type must set for consumer actions.');
+		}
+		this.filterActive = true;
+		this.type = type;
+		this.allowedLogKeys = options.allowedLogKeys || null;
+	}
+
+	/**
+	 * @public
+	 * @returns {Array[string]}
+	 */
+	getSecuredKeys() {
+		return securedKeys;
+	}
+
+	/**
+	 * @public
+	 * @returns {string}
+	 */
+	getType() {
+		return this.type;
+	}
+
+	/**
+	 * @private
+	 * @param {string} type
+	 * @returns {boolean}
+	 */
+	matchType(type) {
+		return this.getType() === type;
+	}
+
+	/**
+	 * @private
+	 * @param {object} data
+	 * @returns {Promise}
+	 */
+	async action(data = {}) {
+		throw new NotImplemented('Must be implemented');
+	}
+
+	/**
+	 * @private
+	 * @param {object} data
+	 * @returns {object}
+	 */
+	filterData(data) {
+		return batchFilterKeys(data, this.allowedLogKeys);
+	}
+
+	/**
+	 * @public
+	 * @param {object} content
+	 * @returns {Promise}
+	 */
+	async exec({ action, data, syncId } = {}) {
+		if (!this.matchType(action)) {
```

Comment:
Data is string, action is type and syncId is identifier for the sync.

Reply:
this action is build as an abstract one. By the caller it can happen that it is registered with one action type, but called with another which is not supported. In this case the exception should be thrown.